iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergFilterFactory.java

```diff
@@ -155,6 +206,92 @@ private static Object leafToLiteral(PredicateLeaf leaf) {
     }
   }
 
+  private static Object getTransformedLiteral(Object literal, TransformSpec transform) {
```

Comment:
can we apply convertLiteral when constructing the Term? see SparkV2Filters#udfToTerm